pkg/webhooks/machine_webhook.go

```diff
@@ -1665,12 +1665,121 @@ func validateNutanix(m *machinev1beta1.Machine, config *admissionConfig) (bool,
 		}
 	}
 
+	// validate gpus if configured
+	if len(providerSpec.GPUs) > 0 {
+		fldPath := field.NewPath("providerSpec", "gpus")
+		for _, gpu := range providerSpec.GPUs {
+			switch gpu.Type {
+			case machinev1.NutanixGPUIdentifierDeviceID:
+				if gpu.DeviceID == nil {
+					errs = append(errs, field.Required(fldPath.Child("deviceID"), "missing gpu deviceID"))
+				}
+			case machinev1.NutanixGPUIdentifierName:
+				if gpu.Name == nil || *gpu.Name == "" {
+					errs = append(errs, field.Required(fldPath.Child("name"), "missing gpu name"))
+				}
+			default:
+				errMsg := fmt.Sprintf("invalid gpu identifier type, the valid types: %q, %q.", machinev1.NutanixGPUIdentifierDeviceID, machinev1.NutanixGPUIdentifierName)
+				errs = append(errs, field.Invalid(fldPath.Child("type"), gpu.Type, errMsg))
+			}
+		}
+	}
+
+	// validate dataDisks if configured
+	if len(providerSpec.DataDisks) > 0 {
+		for _, fldErr := range validateNutanixDataDisks(providerSpec.DataDisks) {
+			errs = append(errs, fldErr)
+		}
+	}
+
 	if len(errs) > 0 {
 		return false, warnings, errs
 	}
 	return true, warnings, nil
 }
 
+func validateNutanixDataDisks(disks []machinev1.NutanixVMDisk) (fldErrs []*field.Error) {
+	fldPath := field.NewPath("providerSpec", "dataDisks")
+	var errMsg string
+
+	for _, disk := range disks {
+		// validate the minimum diskSize of 1Gi
+		diskSizeBytes := disk.DiskSize.Value()
+		if diskSizeBytes < 1024*1024*1024 {
+			fldErrs = append(fldErrs, field.Invalid(fldPath.Child("diskSize"), fmt.Sprintf("%v bytes", diskSizeBytes), "The minimum diskSize is 1Gi bytes."))
+		}
+
+		if disk.DeviceProperties != nil {
+			switch disk.DeviceProperties.DeviceType {
+			case machinev1.NutanixDiskDeviceTypeDisk:
+				if disk.DeviceProperties.AdapterType != machinev1.NutanixDiskAdapterTypeSCSI &&
```

Comment:
You'd probably be able to make this easier to read if you used a switch statement here and had the default case as the error case

Reply:
Changed as suggested.